### PR TITLE
[WIP] #15502 Make template shortcuts be usable without Templating component

### PR DESCRIPTION
**Pull request by Koc**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | ? |
| Fixed tickets | #15502 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by dosten** [comment]
You should add this change to the CHANGELOG.md

**Comment by fabpot** [comment]
:+1:

**Comment by Tobion** [comment]
:+1: but some tiny tests to ensure these methods keep being called would be good.

**Comment by Koc** [comment]
@Tobion can you give advice, maybe some piece of code how to do this? Bootstrap kernel without templating - where should I do it?

**Comment by Tobion** [comment]
I don't mean functional tests but unit tests. We already have unit tests for this class. Just mock the container and ensure the methods are called.

File: src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php [hunk 1]
```diff
@@ -187,11 +201,19 @@ public function render($view, array $parameters = array(), Response $response =
...
+            $templating = $this->container->get('templating');
+
+            $callback = function () use ($templating, $view, $parameters) {
+                $templating->stream($view, $parameters);
+            };
+        } else {
+            $twig = $this->container->get('twig');
+
+            $callback = function () use ($twig, $view, $parameters) {
+                $twig->loadTemplate($view)->display($parameters);
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php]: use ``$twig->display()`` here rather than the internal implementation

Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php]: loadTemplate() is public, so it's not internal

Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php]: But what's the point of using the more verbose version?

Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php]: one less method call ;)
doesn't really matter to me. I'm fine with both

**Comment by dosten** [comment]
And if the `twig` service isn't available? IMO the code should be something like this:

```php
if ($this->container->has('templating')) {
    // use the templating component
} else if ($this->container->has('twig')) {
    // use twig
} else {
    throw new \LogicException('You can not use the XXX method if the Templating Component or the Twig Bundle are not available.');
}
```

**Comment by fabpot** [comment]
Thank you @Koc.

**Comment by fabpot** [comment]
I've reverted the tests

**Pull request closed without merging**